### PR TITLE
refactor 'GlobalButtonOrderGenerator' 

### PR DESCRIPTION
**Pull request by dnqbob**
Modify the `GlobalButtonOrderGenerator`, make `RepairOrderGenerator` and `BeaconOrderGenerator` inherit it properly as an global command for global command panel, as well as makes classes which inherit it more readable&extensible.

### CODE REVIEW

**Comment by dnqbob** [comment]
I guess the orignal version of `RepairOrderGenerator` belongs to `GlobalButtonOrderGenerator`, which is just by the name of `GlobalButtonOrderGenerator`.

However, since @penev92 added units repair function and make `RepairOrderGenerator` outside the box, the name and meaning of this class became confusing and the usage of it became unnecessary.  So I think a refactor can be here to make all global commands for global command panel more readable&extensible.

I put most of common things that global commands has there into `GlobalButtonOrderGenerator` as default settings but can be changed if modders/game devs want. For example, my damnable function of dragging repair&powerdown, @GraionDilach 's idea about YR dragging selling, and probably @MustaphaTR 's idea about sell units on repair pad with a price text renderable. They can also start from inheriting this class to make code more neat and maintainable, instead of inheriting `OrderGenerator` and even `IOrderGenerator` to set a bunch of interface function that may not put into use.

**Comment by abcdefg30** [comment]
This needs a rebase.

**Comment by dnqbob** [comment]
@abcdefg30 
rebased.

File: OpenRA.Mods.Common/Orders/RepairOrderGenerator.cs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#region Copyright & License Information
+﻿#region Copyright & License Information
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Orders/RepairOrderGenerator.cs:1]: Please revert this change.

**Review by abcdefg30 — COMMENTED** [review]
`GlobalButtenOrderGenerator` seems to identical to `OrderGenerator` now, so I'm not sure if this makes much sense. (Maybe `GlobalButtonOrderGenerator` should inherit from `OrderGenerator`?)

**Comment by dnqbob** [comment]
Not going to continue this PR any more, I may do a new PR on make `OrderGenerator` more flexible.
